src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,108 @@
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private User expectedUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.PEOPLE.clear();
+    }
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        expectedUser = new User();
+        expectedUser.setLogin("MyName");
+        expectedUser.setPassword("MyPassword");
+        expectedUser.setAge(18);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_invalidLogin_notOk() {
+        expectedUser.setLogin(null);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(expectedUser);
+        });
+        expectedUser.setLogin("");
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(expectedUser);
+        });
+        expectedUser.setLogin("Abcde");
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(expectedUser);
+        });
+    }
+
+    @Test
+    void register_invalidPassword_notOk() {
+        expectedUser.setPassword(null);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(expectedUser);
+        });
+        expectedUser.setPassword("");
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(expectedUser);
+        });
+        expectedUser.setPassword("abcde");
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(expectedUser);
+        });
+    }
+
+    @Test
+    void register_invalidAge_notOk() {
+        expectedUser.setAge(null);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(expectedUser);
+        });
+
+        expectedUser.setAge(17);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(expectedUser);
+        });
+        expectedUser.setAge(-1);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(expectedUser);
+        });
+    }
+
+    @Test
+    void register_validUser_notOk() {
+        User actualUser = registrationService.register(expectedUser);
+        assertEquals(expectedUser, actualUser);
+        assertNotNull(actualUser.getId());
+    }
+
+    @Test
+    void register_userIsPersoent_notOk() {
+        registrationService.register(expectedUser);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(expectedUser);
+        });

```

Comment:
we cannot use the same method we test in this scenario, add via dao directly
![image](https://github.com/mate-academy/jv-registration-validation/assets/49767049/984d4c0b-2fb9-47a7-8aa9-eb1175eabe72)